tests/integration/test_archives.py

```diff
@@ -0,0 +1,173 @@
+from collections import namedtuple
+
+import psutil
+import pytest
+import vorta.borg
+import vorta.utils
+import vorta.views.archive_tab
+from PyQt6 import QtCore
+from vorta.store.models import ArchiveModel
+
+
+def test_repo_list(qapp, qtbot):
+    main = qapp.main_window
+    tab = main.archiveTab
+
+    main.tabWidget.setCurrentIndex(3)
+    tab.refresh_archive_list()
+    qtbot.waitUntil(lambda: not tab.bCheck.isEnabled(), **pytest._wait_defaults)
+
+    assert not tab.bCheck.isEnabled()
+
+    qtbot.waitUntil(lambda: 'Refreshing archives done.' in main.progressText.text(), **pytest._wait_defaults)
+    assert ArchiveModel.select().count() == 6
+    assert 'Refreshing archives done.' in main.progressText.text()
+    assert tab.bCheck.isEnabled()
+
+
+def test_repo_prune(qapp, qtbot):
+    main = qapp.main_window
+    tab = main.archiveTab
+
+    main.tabWidget.setCurrentIndex(3)
+    tab.refresh_archive_list()
+    qtbot.waitUntil(lambda: tab.archiveTable.rowCount() > 0, **pytest._wait_defaults)
+
+    qtbot.mouseClick(tab.bPrune, QtCore.Qt.MouseButton.LeftButton)
+    qtbot.waitUntil(lambda: 'Refreshing archives done.' in main.progressText.text(), **pytest._wait_defaults)
```

Comment:
Doesn't it say `Refreshing archives done` before the prune already?

Reply:
You are right. Added a check to first check for prune initialiazation.